### PR TITLE
Feature/covid contact biz address

### PR DESCRIPTION
**Pull request by parc-jason**
Model flipped so IPC Plan is top level.
Added location.
Changes to business to have address data.
Handle covid contact.
Update transformations for new UI structure.

### CODE REVIEW

File: app/src/components/validators.js [hunk 1]
```diff
@@ -169,10 +202,84 @@ const validators = {
...
+      errors.push({ value: obj.location['startDate'], message: 'Invalid value for required field `location.startDate`.' });
+    }
+    if (!models.location.endDate(obj.location['endDate'])) {
+      errors.push({ value: obj.location['endDate'], message: 'Invalid value for required field `location.endDate`.' });
+    }
+    if (!models.location.accTents(obj.location['accTents'])) {
+      errors.push({ value: obj.location['accTents'], message: 'Invalid value for required field `location.accTents`.' });
+    } else {
+      if (obj.location['accTents']) {
+        if (!validatorUtils.isNonEmptyString(obj.location['tentDetails'])) {
```
Comment (loneil) [inline app/src/components/validators.js:235]: Might be an open question of whether the details for the tents are required.
If we're requiring it here then I'll force it to be entered on the FE. Unless I'm reading this all bleary-eyed and this isn't a nonempty enforcement.

File: app/src/components/validators.js [hunk 2]
```diff
@@ -169,10 +202,84 @@ const validators = {
...
+          errors.push({ value: obj.location['tentDetails'], message: 'Invalid value for required field `location.tentDetails`.' });
+        }
+      }
+    }
+    if (!models.location.accMotel(obj.location['accMotel'])) {
+      errors.push({ value: obj.location['accMotel'], message: 'Invalid value for required field `location.accMotel`.' });
+    } else {
+      if (obj.location['accMotel']) {
+        if (!validatorUtils.isNonEmptyString(obj.location['motelName'])) {
```
Comment (loneil) [inline app/src/components/validators.js:244]: Same as tent details above

File: app/src/models/business.js [hunk 3]
```diff
@@ -14,17 +14,44 @@ module.exports = (sequelize, DataTypes) => {
...
       unique: false
+    },
+    addressLine1: {
+      allowNull: false,
+      comment: 'Address line 1',
+      type: DataTypes.STRING(255),
+      unique: false
+    },
+    addressLine2: {
+      allowNull: false,
```
Comment (loneil) [inline app/src/models/business.js:25]: AddressLine2 is optional (unless we're just writing empty string if it's not supplied)

File: app/src/models/location.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+        isUUID: 4
+      }
+    },
+    startDate: {
+      allowNull: false,
+      type: DataTypes.DATE
+    },
+    endDate: {
+      allowNull: false,
+      type: DataTypes.DATE
```
Comment (loneil) [inline app/src/models/location.js:18]: I've asked about optionality of these.
Enforcing entry on the front end for now too.

File: app/src/models/business.js (hunk 3 above)
Comment (parc-jason) [reply app/src/models/business.js:25]: good catch..

File: app/src/models/location.js [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+    },
+    addressLine1: {
+      allowNull: false,
+      comment: 'Address line 1',
+      type: DataTypes.STRING(255),
+      unique: false
+    },
+    addressLine2: {
+      allowNull: false,
```
Comment (loneil) [inline app/src/models/location.js:27]: Same as above, optional

File: app/src/models/business.js (hunk 3 above)
Comment (parc-jason) [reply app/src/models/business.js]: fixed

Comment (parc-jason) [reply app/src/models/business.js]: fixed

**Pull request merged**